server/build-osx.sh

```diff
@@ -167,10 +167,10 @@ _build_server_osx() {
 
     # Configure the source tree
     echo "Configuring the postgres source tree for universal binary support"
-    ssh $PG_SSH_OSX "cd $PG_PATH_OSX/server/source/postgres.osx/; PG_SYSROOT=$PG_SYSROOT PATH=/opt/local/Current_v18/bin:$PATH LDFLAGS=\"-L/opt/local/Current_v18/lib\" PYTHON=$PG_PYTHON_OSX/bin/python3 TCL_CONFIG_SH=$PG_TCL_OSX/lib/tclConfig.sh PERL=$PG_PERL_OSX/bin/perl XML2_CONFIG=/opt/local/Current_v18/bin/xml2-config ICU_LIBS=\"-L/opt/local/Current_v18/lib -licuuc -licudata -licui18n\" ICU_CFLAGS=\"-I/opt/local/Current_v18/include\" LZ4_CFLAGS=\"-I/opt/local/Current_v18/include\" LZ4_LIBS=\"-L/opt/local/Current_v18/lib\" ZSTD_CFLAGS=\"-I/opt/local/Current_v18/include\" ZSTD_LIBS=\"-L/opt/local/Current_v18/lib\" LIBCURL_CFLAGS=\"-I/opt/local/Current_v18/include\" LIBCURL_LIBS=\"-L/opt/local/Current_v18/lib\" ./configure --with-icu --enable-debug  --prefix=$PG_STAGING --with-ldap --with-openssl --with-perl --with-python --with-tcl --with-bonjour --with-pam --with-libxml --with-libcurl --with-uuid=e2fs --with-includes=/opt/local/Current_v18/include/libxml2:/opt/local/Current_v18/include:/opt/local/Current_v18/include/security:/opt/local/Current_v18/include/openssl/ --docdir=$PG_STAGING/doc/postgresql --with-libxslt --with-libedit-preferred --with-gssapi --with-lz4 --with-zstd" || _die "Failed to configure postgres for universal support"
+    ssh $PG_SSH_OSX "cd $PG_PATH_OSX/server/source/postgres.osx/; PG_SYSROOT=$PG_SYSROOT PATH=/opt/local/Current_v18/bin:$PATH LDFLAGS=\"-L/opt/local/Current_v18/lib -arch arm64 -arch x86_64\" PYTHON=$PG_PYTHON_OSX/bin/python3 TCL_CONFIG_SH=$PG_TCL_OSX/lib/tclConfig.sh PERL=$PG_PERL_OSX/bin/perl XML2_CONFIG=/opt/local/Current_v18/bin/xml2-config ICU_LIBS=\"-L/opt/local/Current_v18/lib -licuuc -licudata -licui18n\" ICU_CFLAGS=\"-I/opt/local/Current_v18/include -arch arm64 -arch x86_64\" LZ4_CFLAGS=\"-I/opt/local/Current_v18/include -arch arm64 -arch x86_64\" LZ4_LIBS=\"-L/opt/local/Current_v18/lib\" ZSTD_CFLAGS=\"-I/opt/local/Current_v18/include -arch arm64 -arch x86_64\" ZSTD_LIBS=\"-L/opt/local/Current_v18/lib\" LIBCURL_CFLAGS=\"-I/opt/local/Current_v18/include -arch arm64 -arch x86_64\" LIBCURL_LIBS=\"-L/opt/local/Current_v18/lib\" ./configure --with-icu --enable-debug --prefix=$PG_STAGING --with-ldap --with-openssl --with-perl --with-python --with-tcl --with-bonjour --with-pam --with-libxml --with-libcurl --with-uuid=e2fs --with-includes=/opt/local/Current_v18/include/libxml2:/opt/local/Current_v18/include:/opt/local/Current_v18/include/security:/opt/local/Current_v18/include/openssl/ --docdir=$PG_STAGING/doc/postgresql --with-libxslt --with-libedit-preferred --with-gssapi --with-lz4 --with-zstd CFLAGS=\"$PG_ARCH_OSX_CFLAGS -O2 -arch arm64 -arch x86_64\"" || _die "Failed to configure postgres for universal support"
 
```

Comment:
I wonder why not just "CFLAGS=\"$PG_ARCH_OSX_CFLAGS" is enough to build universal binary